monkestation/code/modules/art_sci_overrides/activators/range.dm

```diff
@@ -0,0 +1,51 @@
+//range artifacts require stimuli to fall within a range between amount and upper range
+// hint range and hint chance are added onto range to see if something we should pull a hint for the user
+
+/datum/artifact_activator/range
+	name = "Generic Range Trigger"
+	//the upper range of the weapon basically between amount, and upper_range
+	var/upper_range = 0
+	///Hint range goes like amount - hint_range to upper_range + hint_range
+	var/hint_range = 0
+	///if we are in the hint range the odds of pulling a hint out.
+	var/hint_prob = 15
+
+/datum/artifact_activator/range/setup(potency)
+	. = ..()
+	upper_range = amount + (hint_range * 2)
+
+/datum/artifact_activator/range/force
+	name = "Physical Trauma"
+	required_stimuli = STIMULUS_FORCE
+	highest_trigger_amount = 30 //any higher than this and its gonna be practically impossible to trigger
+	hint_prob = 50
+	hint_range = 10
+	hint_texts = list("you almost want to start hitting things.", "a good whack might fix this.")
+
+/datum/artifact_activator/range/heat
+	name = "Heat Sensisty"
+	required_stimuli = STIMULUS_HEAT
+	hint_range = 20
```

Comment:
might want to make this hint range be higher with how large the highest_trigger_amount is.